### PR TITLE
fix(arc-303): implement the determine user group flowchart

### PR DESCRIPTION
**Pull request by bartnaessens**
CleanShot 2022-04-05 at 17 14 33

### CODE REVIEW

File: src/modules/users/types.ts [hunk 1]
```diff
@@ -50,3 +52,17 @@ export interface GqlPermission {
 export interface GqlPermissionData {
 	permission: GqlPermission;
 }
+
+export enum Group {
+	KIOSK_VISITOR = '04150e6e-b779-4125-84e5-6ee6fc580757',
+	MEEMOO_ADMIN = '0b281484-76cd-45a9-b6ce-68a0ea7f4b26',
+	VISITOR = '0213c8d4-f459-45ef-8bbc-96268ab56d01',
+	CP_ADMIN = 'c56d95aa-e918-47ca-b102-486c9449fc4a',
+}
```
Comment (ian-emsens-shd) [inline src/modules/users/types.ts:61]: Is hier een bijhorende PR op de `hasura` repo voor of zitten deze al vast in de seeders?

File: src/modules/users/types.ts [hunk 2]
```diff
@@ -50,3 +52,17 @@ export interface GqlPermission {
...
+
+export enum GroupIdToName {
+	'04150e6e-b779-4125-84e5-6ee6fc580757' = 'KIOSK_VISITOR',
+	'0b281484-76cd-45a9-b6ce-68a0ea7f4b26' = 'MEEMOO_ADMIN',
+	'0213c8d4-f459-45ef-8bbc-96268ab56d01' = 'VISITOR',
+	'c56d95aa-e918-47ca-b102-486c9449fc4a' = 'CP_ADMIN',
+}
```
Comment (ian-emsens-shd) [inline src/modules/users/types.ts:68]:
```suggestion
export const GroupIdToName = {
	[Group.KIOSK_VISITOR]: 'KIOSK_VISITOR',
	[Group.MEEMOO_ADMIN]: 'MEEMOO_ADMIN',
	[Group.VISITOR]: 'VISITOR',
	[Group.CP_ADMIN]: 'CP_ADMIN',
};
```

**Review by ian-emsens-shd — APPROVED** [review]

File: src/modules/users/types.ts (hunk 1 above)
Comment (bartnaessens) [reply src/modules/users/types.ts:61]: Deze komen al uit de seeds

File: src/modules/auth/services/idp.service.ts [hunk 3]
```diff
@@ -24,4 +35,40 @@ export class IdpService {
...
+	/**
+	 * Flowchart: see https://meemoo.atlassian.net/wiki/spaces/HA2/pages/2978447456/Gebruikersgroepen+en+permissies+BZT+versie+1#Functionele-samenvatting
+	 */
+	public async determineUserGroup(ldapUser: LdapUser): Promise<Group> {
+		// permissions check
+		const apps = get(ldapUser, 'attributes.apps', []);
+		if (
+			!apps.includes('hetarchief') &&
+			!apps.includes('admins') // TODO replace by a single value 'hetarchief-beheer' once archief 2.0 is launched
+		) {
```
Comment (bertyhell) [inline src/modules/auth/services/idp.service.ts:48]: i would flip this IF check. It's easier to reason about "true" statements than it is to reason about not (!) statements
```
if (
apps.includes('hetarchief') ||
apps.includes('admins') // TODO replace by a single value 'hetarchief-beheer' once archief 2.0 is launched
) {
  // Handle admin logic
}
// Handle visitor/kiosk logic
	
```

File: src/modules/auth/services/idp.service.ts [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
+import { Idp, LdapUser } from '~shared/auth/auth.types';
 
 @Injectable()
 export class IdpService {
+	private logger: Logger = new Logger(IdpService.name, { timestamp: true });
 	private idpsWithSpecificLogoutPage = [Idp.HETARCHIEF, Idp.MEEMOO];
 
-	constructor(protected configService: ConfigService) {}
+	public static MEEMOO_ORGANISATION_ID = 'OR-w66976m';
+	public static SHD_ORGANISATION_ID = 'OR-pv6b877';
```
Comment (bertyhell) [inline src/modules/auth/services/idp.service.ts:18]: should we expose these as environment variables? => i'll ask on the tech sync

**Review by bertyhell — APPROVED** [review]

File: src/modules/auth/services/idp.service.ts (hunk 3 above)
Comment (bartnaessens) [reply src/modules/auth/services/idp.service.ts]: fixed

File: src/modules/auth/services/idp.service.ts (hunk 4 above)
Comment (bartnaessens) [reply src/modules/auth/services/idp.service.ts]: extracted to env vars

**Pull request merged**